lib/ansible/module_utils/vmware.py

```
@@ -213,6 +213,16 @@ def find_host_portgroup_by_name(host, portgroup_name):
     return None
 
 
+def _get_vm_prop(vm, expression):
+    """Safely get a property or return None"""
+    result = None
+    try:
+        result = eval(expression)
+    except:
+        pass
+    return result
```

Something like this instead:
``` python
def _get_vm_prop(vm, attributes):
    result = vm
    for attribute in attributes:
        try:
            result = getattr(result, attribute)
        except AttributeError:
            return None
    return result

_get_vm_props(vm, ('guest', 'toolsRunningStatus'))
```

---

I think, using `pass` instead of returning `None` in AttributeError as `return None` fails to retrieve snapshots. Snapshot data works perfect with `pass`.

---

pass doesn't advance result to the next layer of the object, though.  Imagine a data structure like this:
``` python
class Alice:
    foo = 1

class Bob:
     bar = Alice()
     foo = 2

obj = Bob()
del obj.bar
result = _get_vm_props(obj, ('bar', 'foo')
# with return None:
result == None
# with pass
result == 2
```

For snapshot, what are you calling?  This seems right and it seems like it would do the right thing:
```
snapshot = _get_vm_prop(vm, ['snapshot'])
```

---

@Akasurde I've tested this patch on cloning from snapshots, so I think it does work to some degree. Can you extrapolate on which situations you don't think it will work?